File: resilient/resilient-annotation-processor/src/main/java/io/koraframework/resilient/annotation/processor/aop/RateLimitKoraAspectFactory.java
package io.koraframework.resilient.annotation.processor.aop;

import io.koraframework.aop.annotation.processor.KoraAspect;
import io.koraframework.aop.annotation.processor.KoraAspectFactory;

import javax.annotation.processing.ProcessingEnvironment;
import java.util.Optional;

public class RateLimitKoraAspectFactory implements KoraAspectFactory {

    @Override
    public Optional<KoraAspect> create(ProcessingEnvironment processingEnvironment) {
        return Optional.of(new RateLimitKoraAspect(processingEnvironment));
    }
}

File: resilient/resilient-symbol-processor/src/main/kotlin/io/koraframework/resilient/symbol/processor/aop/RateLimitKoraAspectFactory.kt
package io.koraframework.resilient.symbol.processor.aop

import com.google.devtools.ksp.KspExperimental
import com.google.devtools.ksp.processing.Resolver
import io.koraframework.aop.symbol.processor.KoraAspect
import io.koraframework.aop.symbol.processor.KoraAspectFactory

@KspExperimental
class RateLimitKoraAspectFactory : KoraAspectFactory {
    override fun create(resolver: Resolver): KoraAspect = RateLimitKoraAspect(resolver)
}
